wolvic/browser/vr/wvr_manager.cc

```diff
@@ -403,15 +403,24 @@ WvrManager::GetInputSourceState() {
         controller.hand == mozilla::gfx::ControllerHand::Left
             ? device::mojom::XRHandedness::LEFT
             : device::mojom::XRHandedness::RIGHT;
-
     // TODO: Get from external
     input_source->description->profiles = {
         "oculus-touch-v3", "oculus-touch-v2", "oculus-touch",
         "generic-trigger-squeeze-thumbstick"};
 
-    gfx::Transform input_from_pointer;
-    WvrMatToTransform(controller.axisValue, &input_from_pointer);
-    input_source->description->input_from_pointer = input_from_pointer;
+    input_source->gamepad = device::Gamepad();
+    input_source->gamepad->buttons_length = controller.numButtons;
+    input_source->gamepad->hand =
+        controller.hand == mozilla::gfx::ControllerHand::Left
+            ? device::GamepadHand::kLeft
+            : device::GamepadHand::kRight;
```

Comment:
For this kind of conversions between mozilla and chrome structures, I would like to see explicit convert methods (that can be inlined in the future if we worry about performance).

Something like:

device::GamepadHand ToGamepadHand(mozilla::gfx::ControllerHand moz_hand) {
  switch(moz_hand) {
    ...
}

Speciall if this is used from several files. I don't want to have the explicit evaluation everywhere.

Same with other conversions we are inlining in this method, making it too much complex. I would prefer helper methods for each of the mozilla to chrome structure conversions (i.e. handedness, targetRayMode).

In those methods we can also have specific warnings for the case a new value comes from Mozilla side we don't know about.

Reply:
I refactored this conversion using the suggestions from @jdapena. However, the conversion above is using XRHandedness instead of device::GamepadHand so this will have to be a separate method. I didn't touch the rest of the code in this change but I will submit a several PR refactoring the rest of the conversion code in a similar way.